File: JAVA/Shirt.java
/* Saya Alfen Fajri Nurulhaq (2201431) mengerjakan Latihan Prakitum 2 OOP class & encapsulation dalam mata
   kuliah Strukdat untuk keberkahanNya maka saya tidak
   melakukan kecurangan seperti yang telah dispesikasikan.Aamiin*/

// Mendefinisikan kelas Shirt sebagai subclass dari class Clothing dan Superclass Product
class Shirt extends Clothing {
    // Mendeklarasikan variabel privat untuk color dan sleeve_type
    private String color;
    private String sleeve_type;

    // Konstruktor default tanpa parameter
    public Shirt() {
        // Inisialisasi nilai default untuk variabel color dan sleeve_type
        this.color = "";
        this.sleeve_type = "";
    }

    // Konstruktor dengan parameter untuk inisialisasi nilai awal
    public Shirt(int idProduct, String name, String brand, int price, String size, String material, String gender, String color, String sleeve_type) {
        // Memanggil konstruktor superclass (Clothing) dengan menggunakan kata kunci "super"
        super(size, material, gender);

        // Menginisialisasi variabel dengan nilai dari parameter konstruktor
        this.color = color;
        this.sleeve_type = sleeve_type;

        // Mengatur nilai variabel dari superclass (Product)
        setIdProduct(idProduct);
        setName(name);
        setBrand(brand);
        setPrice(price);
    }

    // Metode untuk mendapatkan nilai color
    public String getColor() {
        return color;
    }

    // Metode untuk mengatur nilai color
    public void setColor(String color) {
        this.color = color;
    }

    // Metode untuk mendapatkan nilai sleeve_type
    public String getSleeve_type() {
        return sleeve_type;
    }

    // Metode untuk mengatur nilai sleeve_type
    public void setSleeve_type(String sleeve_type) {
        this.sleeve_type = sleeve_type;
    }
}

File: PYTHON/Shirt.py
# Saya Alfen Fajri Nurulhaq (2201431) mengerjakan Latihan Prakitum 2 OOP
# Inheritance dalam mata
# kuliah Strukdat untuk keberkahanNya maka saya tidak
# melakukan kecurangan seperti yang telah dispesikasikan.Aamiin

from Clothing import Clothing

# Mendefinisikan kelas Shirt sebagai subclass dari class Clothing dan Superclass Product
class Shirt(Clothing):
    # Mendeklarasikan variabel privat untuk color dan sleeve_type
    def __init__(self, idProduct=0, name="", brand="", price=0, size="", material="", gender="", color="", sleeve_type=""):
        # Memanggil konstruktor superclass (Clothing)
        super().__init__(size, material, gender)

        # Menginisialisasi variabel dengan nilai dari parameter konstruktor
        self.color = color
        self.sleeve_type = sleeve_type

        # Mengatur nilai variabel dari superclass (Product)
        self.set_id_product(idProduct)
        self.set_name(name)
        self.set_brand(brand)
        self.set_price(price)

    # Metode untuk mendapatkan nilai color
    def get_color(self):
        return self.color

    # Metode untuk mengatur nilai color
    def set_color(self, color):
        self.color = color

    # Metode untuk mendapatkan nilai sleeve_type
    def get_sleeve_type(self):
        return self.sleeve_type

    # Metode untuk mengatur nilai sleeve_type
    def set_sleeve_type(self, sleeve_type):
        self.sleeve_type = sleeve_type
